src/commands/registries/azure/DockerWebhookCreateStep.ts

```diff
@@ -0,0 +1,82 @@
+        super();
+        this._treeItem = treeItem;
+    }
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{
+        message?: string;
+        increment?: number;
+    }>): Promise<void> {
+        const site: Site = nonNullProp(context, 'site');
+        let siteClient = new SiteClient(site, context);
+        let appUri: string = (await siteClient.getWebAppPublishCredential()).scmUri;
+        if (this._treeItem.parent instanceof AzureRepositoryTreeItem) {
+            const creatingNewApp: string = `Creating webhook for webapp "${context.newSiteName}"...`;
+            ext.outputChannel.appendLine(creatingNewApp);
+            progress.report({ message: creatingNewApp });
+            const webhook = await this.createWebhookForApp(this._treeItem, context, appUri);
+            ext.outputChannel.appendLine(`Created webhook '${webhook.name}' with scope '${webhook.scope}', id: '${webhook.id}' and location: '${webhook.location}'`);
+        } else if (this._treeItem.parent instanceof DockerHubRepositoryTreeItem) {
+            // point to dockerhub to create a webhook
+            // http://cloud.docker.com/repository/docker/<registryName>/<repoName>/webHooks
+            const dockerhubPrompt: string = "Copy web app endpoint and browse to dockerhub";
+            let response: string = await vscode.window.showInformationMessage("Please browse to your dockerhub account to set up a CI/CD webhook", dockerhubPrompt);
+            if (response) {
+                vscode.env.clipboard.writeText(appUri);
+                // tslint:disable-next-line: no-floating-promises
+                openExternal(`https://cloud.docker.com/repository/docker/${this._treeItem.parent.parent.parent.username}/${this._treeItem.parent.repoName}/webHooks`);
+            }
+        }
+    }
+
+    public shouldExecute(context: IAppServiceWizardContext): boolean {
+        return !!context.site && (this._treeItem.parent instanceof AzureRepositoryTreeItem || this._treeItem.parent instanceof DockerHubRepositoryTreeItem);
+    }
+
+    private async createWebhookForApp(node: RemoteTagTreeItem, wizardContext: IActionContext & Partial<IAppServiceWizardContext>, appUri: string): Promise<Webhook | undefined> {
+        // fields derived from the app service wizard
+        let siteName: string = wizardContext.site.name;
+        let baseName = `webapp${siteName}`;
+        let webhookName: string = baseName;
+        // variables derived from the container registry
+        const registryTreeItem: AzureRegistryTreeItem = (<AzureRepositoryTreeItem>node.parent).parent;
+        const crmClient = createAzureClient(registryTreeItem.parent.root, ContainerRegistryManagementClient);
+        const existingWebhooks = await crmClient.webhooks.list(registryTreeItem.resourceGroup, registryTreeItem.registryName);
+        let dedupeCount: number = 0;
+        while (existingWebhooks.find((hook) => hook.name === webhookName)) {
```

Comment:
The way it is now, wouldn't create a new webhook every time you deploy to Azure?  You could probably check the scope of the webhook and verify it doesn't match the scope you are creating.

Also, I'd have a dedupeCount max otherwise there's a potential (albeit low) chance that you'll hit an infinite while loop.

Reply:
I feel that the naming needs to be updated. Deploy to Azure doesn't merely deploy, but it also creates a web app. 

The ideal flow is that you deploy only once. Anytime you make a change to your code and build your container, pushing to ACR should update the web app (deploy is done separately)

Reply:
Okay, but the webhooks are all listed by registry rather than by web app then?

Reply:
Yes. The registry owns the webhooks, and the tier of the registry decides how many webhooks you can have, etc.